Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.cpp

```diff
@@ -0,0 +1,475 @@
+    const char* ScriptAdapterComponent::GEN_DIR_NAME = "gen";
+    const char* ScriptAdapterComponent::GLOBAL_LUA_FILE_NAME = "GlobalsStaticLuaData.lua";
+    const char* ScriptAdapterComponent::CLASS_LUA_FILE_NAME = "ClassesStaticLuaData.lua";
+    const char* ScriptAdapterComponent::EBUS_LUA_FILE_NAME = "EbusesStaticLuaData.lua";
+    const int ScriptAdapterComponent::NAME_BUFFER_SIZE = 64;
+
+    void ScriptAdapterComponent::Reflect(ReflectContext* context)
+    {
+        SerializeContext* serializeContext = azrtti_cast<SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<ScriptAdapterComponent, Component>()->Version(1);
+        }
+    }
+
+    void ScriptAdapterComponent::Activate()
+    {
+        AdapterBus::Handler::BusConnect();
+    }
+
+    void ScriptAdapterComponent::Deactivate()
+    {
+        AdapterBus::Handler::BusDisconnect();
+    }
+
+    AZStd::string ScriptAdapterComponent::GetProjectPath()
+    {
+        ComponentApplication* app = nullptr;
+        EBUS_EVENT_RESULT(app, ComponentApplicationBus, GetApplication);
+        const char* engineRoot = app->GetEngineRoot();
+        AZ_Assert(engineRoot != nullptr, "app GetEngineRoot failed");
+        const auto projectName = Utils::GetProjectName();
+        return AZStd::string::format("%s\\%s\\", engineRoot, projectName.c_str());
+    }
+
+    AZStd::string ScriptAdapterComponent::GetAbsoluteLuaDirPath()
+    {
+        return AZStd::string::format("%s%s", GetProjectPath().c_str(), RELATIVE_LUA_PATH);
+    }
+
+    AZStd::string ScriptAdapterComponent::GetGenDirPath()
+    {
+        return AZStd::string::format("%s%s", GetProjectPath().c_str(), GEN_DIR_NAME);
+    }
+
+    bool ScriptAdapterComponent::SetGenDirWritable(bool isWritable)
+    {
+    #if defined(WIN32) || defined(WIN64)
+        char userNameBuffer[NAME_BUFFER_SIZE];
+        DWORD userNameSize = NAME_BUFFER_SIZE;
+        if(!GetUserName(userNameBuffer, &userNameSize)) {
+            return false;
+        }
+        CDacl dacl;
+        CSid sid;
+        // bind user identifier
+        if (!sid.LoadAccount(userNameBuffer)) {
+            return false;
+        }
+        AZStd::string genPath = GetGenDirPath();
+        if (!AtlGetDacl(genPath.c_str(), SE_FILE_OBJECT, &dacl)) {
+            return false;
+        }
+
+        dacl.RemoveAllAces();
+        dacl.AddAllowedAce(sid, GENERIC_READ | GENERIC_EXECUTE, OBJECT_INHERIT_ACE | CONTAINER_INHERIT_ACE);
+        if (!isWritable)
+        {
+            dacl.AddDeniedAce(sid, DELETE | FILE_DELETE_CHILD | FILE_WRITE_DATA | FILE_APPEND_DATA | FILE_WRITE_EA | FILE_WRITE_ATTRIBUTES, OBJECT_INHERIT_ACE | CONTAINER_INHERIT_ACE);
+        }
+
+        return AtlSetDacl(genPath.c_str(), SE_FILE_OBJECT, dacl) ? true : false;
+    #elif defined(LINUX)
+        SetGenDirWritableRecursive(GetGenDirPath().c_str(), isWritable);
+        return true;
+    #endif
+    }
```

Comment:
This code needs to be modified to use the platform abstraction layer (PAL)

https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#code

Code
Required: All platform specific code needs to be implemented according to the patterns defined by the PAL effort (Platform Abstraction Layer).

Avoid adding platform specific defines to the codebase.